lib/errors.ts

```diff
@@ -23,19 +23,36 @@ import { getHelp } from './utils/messages';
 
 export class ExpectedError extends TypedError {}
 
-export class NotLoggedInError extends ExpectedError {}
+// Note: Because of a nasty TS limitation re extending builtins, instanceof was not working correctly with
+// subclassed errors.  The workaround is to set their prototypes manually.
+// https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+// https://github.com/microsoft/TypeScript/issues/13965
```

Comment:
Ouch. Could we at least do the `setPrototypeOf` trick to the `ExpectedError` class, and then avoid having to do it for subclasses of `ExpectedError`?

Also, I wonder if there is something that [our own `instanceOf` function](https://github.com/balena-io/balena-cli/blob/v12.22.2/lib/errors.ts#L65) could do work better than the built-in `instanceof` in this regard. (Even if there was though, we might still want/need the `setPrototypeOf` trick.)

Reply:
Do you mean just set it once in ExpectedError? No I don't think that will work according to MS advisory.

Reply:
"However, any subclass of FooError will have to manually set the prototype as well."
https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

Reply:
I thought about adding to the instanceOf function, but I don't think there's anything it can test. (I guess if there were, then the compiler could implement this, and fix the situation)

Reply:
Double ouch. Does `TypedErorr` (npm `typed-error` package) implement these `setPrototypeOf` workarounds? Because the functionality provided by `TypedError` might be broken too, and presumably we rely on something they provide. It's even tempting not to subclass the built-in classes, though we might end up with `instanceof Error` returning `false`. :-/

Reply:
I had a quick look before, and I'm not sure - it does set the prototype but not sure if it's to this effect.

@Page- I think you were a contributor to typed-error.  Could you chime in?

Reply:
It's doing `Object.setPrototypeOf(this, new.target.prototype)`

Reply:
`typed-error` isn't affected by this because it's either an old version that did manual extension (via a somewhat convoluted es5 method) or it's a modern version that requires es6 and doesn't rely on typescript downleveling so these `Object.setPrototypeOf` changes are unnecessary